File: frontend/iosApp/UseCases/Sources/UseCases/DomainMappers/UserMapper.swift
import Foundation
import Domains
import APIGateways
import Utilities

public enum UserMapper {
    public static func toDomain(_ response: UserResponse) -> User {
        User(
            id: response.id,
            name: response.name,
            username: response.username,
            birthday: response.birthday.flatMap { DateFormatters.yyyyMMdd.date(from: $0) },
            avatarUrl: response.avatarUrl.flatMap { URL(string: $0) }
        )
    }
}

File: frontend/shared/src/commonMain/kotlin/com/example/memoriesapp/mapper/UserMapper.kt
package com.example.memoriesapp.mapper

import com.example.memoriesapp.api.response.UserResponse
import com.example.memoriesapp.core.SyncStatus
import com.example.memoriesapp.domain.User
import kotlinx.datetime.LocalDate

/**
 * Mapper for User domain model
 */
object UserMapper {
    /**
     * Convert API response to domain model
     */
    fun toDomain(response: UserResponse): User {
        return User(
            id = response.id,
            name = response.name,
            username = response.username,
            birthday = response.birthday?.let { parseBirthday(it) },
            avatarUrl = response.avatarUrl,
            avatarLocalPath = null,
            syncStatus = SyncStatus.SYNCED
        )
    }

    private fun parseBirthday(dateString: String): LocalDate? {
        return try {
            LocalDate.parse(dateString)
        } catch (e: Exception) {
            null
        }
    }
}
